### ISSUE TITLE
Adding support for `RUSTFLAGS`

### ISSUE THREAD

**Issue by elsandosgrande** [vaeth/portage-bashrc-mv#33]
This ticket is meant more to see if the maintainer is willing for the list of `FLAGS` variables to be expanded more so than a request for the maintainer himself to implement said expansion.

Basically, I would like to at least try and implement support for `RUSTFLAGS` (which was not present the last time I checked, though I could be mistaken) at a slightly later date once I have ample free time to pour into the effort. However, I would first like to know if the addition would be accepted (assuming the code quality is up to snuff and all) before going about doing it.

Have a nice day!

**Comment by vaeth** [comment]
I use currently only a few rust packages, but I am afraid that I might more and more be forced to change this, so adding support might make sense.

However, currently, I have no experience with `RUSTFLAGS` (e.g. can their arguments be separated with `=`? is there something like `LDRUSTFLAGS` with modified syntax? etc): In the beginning would have to completely rely on PRs.

Sure, I would include reasonable PRs (not only for `RUSTFLAGS`). Would be nice if you could also update `bashrc.d/README` correspondingly (and possibly also `README.md` and/or `AUTHORS`, `NEWS` though I might modify the latter e.g. to decide about version number) - especially for new features or behavior changes this is usually the most time-consuming work...

**Comment by elsandosgrande** [comment]
I'm glad to see that you're not completely opposed to the proposal!

Regarding your questions in the second paragraph, I'm not quite sure what you mean by “[…] e.g. can their arguments be separated by `=`? […]” :sweat_smile:. I'm still learning Rust, but my impression is that `RUSTFLAGS` only get passed to rustc, not Cargo, so the only valid flags are those described [here](https://doc.rust-lang.org/rustc/command-line-arguments.html). The general syntax is `RUSTFLAGS="[flags]"`, where the code generation (codegen) options specifically are `RUSTFLAGS="-C key1=value1 -C key2=value2 -C keyn=valuen"`; I haven't used `RUSTFLAGS` for anything other than code generation options, so my experience here is limited at the moment :sweat_smile:. In case you're wondering, the space after `-C` is not required, nor is `=value` for whichever code generation option [the documentation](https://doc.rust-lang.org/rustc/codegen-options/index.html) includes “[…] or no value: […]”; for example, to enable FatLTO, which is the default form of link-time optimisation, you can pass `-Clto` instead of `-Clto=true` or `-Clto=fat`. Also, there are no separate linker flags as far as I know, they're just additional code generation options for rustc.

---

Here are my current `RUSTFLAGS` for reference:
```
RUSTFLAGS="-C strip=none -C target-cpu=native -C opt-level=3 -C debuginfo=2 -C codegen-units=1 -C link-arg=-fuse-ld=mold -C linker-plugin-lto=/usr/lib/llvm/16/lib64/LLVMgold.so -C lto=thin -C embed-bitcode -C llvm-args=--polly -C llvm-args=--polly-vectorizer=stripmine"
```

---

Regarding documentation, that's actually going to be the easy part, as I'm not all too familiar with shell code at the moment :sweat_smile:. I mean, I've written a few large BASH scripts, but it's been at least a year since then, so, it's going to take a bit for me to get going with this; the two reasons why I'm not starting right away are that I'm still moving over to Clang/LLVM as my global compiler and that my mum and I currently have our hands full with taking care of granny due to medical reasons, which has delayed the transition to Clang by about a month now. Also, I would prefer not to touch `NEWS` and other Gentoo-specific documentation, as I'm not yet familiar with how Gentoo packages work in full :sweat_smile:; the `README`s are fine though.

**Comment by vaeth** [comment]
Please try whether you can use e.g.

```
RUSTFLAGS="-C=strip=none -C=target-cpu=native -C=opt-level=3 -C=debuginfo=2 -C=codegen-units=1 -C=link-arg=-fuse-ld=mold -C=linker-plugin-lto=/usr/lib/llvm/16/lib64/LLVMgold.so -C=lto=thin -C=embed-bitcode -C=llvm-args=--polly -C=llvm-args=--polly-vectorizer=stripmine
```

It would be much simpler to support only space separated flags (for instance, if you want to filter out such a flag, do not attempt to filter out something with several parameters): This happens also for the existing *FLAGS, though unfortunately, a few exceptions had to be made.

**Comment by elsandosgrande** [comment]
Yeah, that doesn't work.

**Comment by vaeth** [comment]
I released 22.0 with support for RUSTFLAGS.
